crates/scroll/state-commitment/src/account.rs

```diff
@@ -36,3 +39,31 @@ impl From<(Account, B256)> for ScrollTrieAccount {
         }
     }
 }
+
+impl From<GenesisAccount> for ScrollTrieAccount {
+    fn from(account: GenesisAccount) -> Self {
+        let storage_root = account
+            .storage
+            .map(|storage| {
+                crate::storage_root_unhashed(
+                    storage
+                        .into_iter()
+                        .filter(|(_, value)| !value.is_zero())
+                        .map(|(slot, value)| (slot, U256::from_be_bytes(*value))),
+                )
+            })
+            .unwrap_or(reth_scroll_primitives::poseidon::EMPTY_ROOT_HASH);
+
+        Self {
+            nonce: account.nonce.unwrap_or_default(),
+            balance: account.balance,
+            storage_root,
+            code_hash: account.code.as_ref().map_or(KECCAK_EMPTY, keccak256),
+            poseidon_code_hash: account
+                .code
+                .as_ref()
+                .map_or(POSEIDON_EMPTY, reth_scroll_primitives::poseidon::hash_code),
+            code_size: account.code.map(|c| c.len()).unwrap_or_default() as u64,
+        }
+    }
+}
```

Comment:
Why do we need this?

Reply:
this is used [here](https://github.com/scroll-tech/reth/pull/82/files#diff-8e95e1ac11fcfab6e18b8182c4b4889cda75be5209754874f0b29a8a2725e7efR240-R241), which enforces a `Into<ScrollTrieAccount>` bound on the `state_root_ref_unhashed` call.